LgiHdmiCec/LgiHdmiCec.cpp

```diff
@@ -0,0 +1,1190 @@
+*
+* Copyright 2019 RDK Management
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+#include <mutex>
+
+#include "LgiHdmiCec.h"
+
+
+#include "ccec/Connection.hpp"
+#include "ccec/CECFrame.hpp"
+#include "host.hpp"
+#include "ccec/host/RDK.hpp"
+
+#include "ccec/drivers/iarmbus/CecIARMBusMgr.h"
+
+
+#include "dsMgr.h"
+#include "dsDisplay.h"
+#include "videoOutputPort.hpp"
+#include "manager.hpp"
+
+#include "websocket/URL.h"
+
+#include "utils.h"
+
+#define HDMICEC_METHOD_SET_ENABLED "setEnabled"
+#define HDMICEC_METHOD_GET_ENABLED "getEnabled"
+#define HDMICEC_METHOD_GET_CEC_ADDRESSES "getCECAddresses"
+#define HDMICEC_METHOD_SEND_MESSAGE "sendMessage"
+#define HDMICEC_METHOD_ENABLE_ONE_TOUCH_VIEW "enableOneTouchView"
+#define HDMICEC_METHOD_TRIGGER_ACTION "triggerAction"
+#define HDMICEC_METHOD_SET_PING_INTERVAL "setPingInterval"
+#define HDMICEC_METHOD_GET_CONNECTED_DEVICES "getConnectedDevices"
+#define HDMICEC_METHOD_SET_NAME "setName"
+#define HDMICEC_METHOD_SET_ONE_TOUCH_VIEW_POLICY "setOneTouchViewPolicy"
+
+#define HDMICEC_EVENT_ON_DEVICES_CHANGED "onDevicesChanged"
+#define HDMICEC_EVENT_ON_MESSAGE "onMessage"
+#define HDMICEC_EVENT_ON_HDMI_HOT_PLUG "onHdmiHotPlug"
+#define HDMICEC_EVENT_ON_CEC_ADDRESS_CHANGE "cecAddressesChanged"
+
+#define PHYSICAL_ADDR_CHANGED 1
+#define LOGICAL_ADDR_CHANGED 2
+#define DEV_TYPE_TUNER 1
+
+#define HDMI_HOT_PLUG_EVENT_CONNECTED 0
+#define HDMI_HOT_PLUG_EVENT_DISCONNECTED 1
+
+#if defined(HAS_PERSISTENT_IN_HDD)
+#define CEC_SETTING_ENABLED_FILE "/tmp/mnt/diska3/persistent/ds/cecData.json"
+#elif defined(HAS_PERSISTENT_IN_FLASH)
+#define CEC_SETTING_ENABLED_FILE "/opt/persistent/ds/cecData.json"
+#else
+#define CEC_SETTING_ENABLED_FILE "/opt/ds/cecData.json"
+#endif
+
+#define CEC_SETTING_ENABLED "cecEnabled"
+
+namespace
+{
+    using namespace WPEFramework;
+
+    inline void requestRescanning(int id)
+    {
+        IARM_Bus_CECHost_ConfigureScan_Param_t param  {};
+        param.needFullUpdate = true;
+        param.scanReasonId = id;
+
+        const IARM_Result_t ret = IARM_Bus_Call(IARM_BUS_CECHOST_NAME,
+                                                IARM_BUS_CEC_HOST_ConfigureScan,
+                                                (void *)&param,sizeof(param));
+
+        if (IARM_RESULT_SUCCESS != ret)
+        {
+            LOGERR("%s failed result %d", IARM_BUS_CEC_HOST_ConfigureScan, ret);
+        }
+    }
+    inline void setOsdName(const string& name)
```

Comment:
I would prefere same naming convention in entire file:
setOsdName vs setOSDName
This may be confusing